bin/add-cmderplugin.ps1

```diff
@@ -0,0 +1,161 @@
+<#
+.Synopsis
+    Add a plugin to Cmder
+.DESCRIPTION
+    Use this script to add a plugin the currently running Cmder
+
+    This script downloads and unpacks sources, configures users aliases, and runs pre and post commands defined in the supplied cmder-plugin-[name].json file.
+
+    You will need to make this script executable by setting your Powershell Execution Policy to Remote signed.
+
+    Then unblock the script for execution with UnblockFile .\add-cmderplugin.ps1
+.EXAMPLE
+    .\add-cmderplugin.ps1 -PluginPath '~/custom/cmder-plugin-[name].json'
+
+    Reads the source json data and excutes the steps to install nad configue the plugin.
+.EXAMPLE
+    .\add-cmderplugin.ps1 -PluginPath '~/custom/cmder-plugin-[name].json' -verbose
+
+    Reads the source json data and excutes the steps to install nad configue the plugin and see more detail as to what is going on.
+.NOTES
+    AUTHORS
+    Dax Games, Samuel Vasko, Jack Bennett
+    Part of the Cmder project.
+.LINK
+    http://cmder.net/ - Project Home
+#>
+
+[CmdletBinding(SupportsShouldProcess=$true)]
+Param(
+    # CmdletBinding will give us;
+    # -verbose switch to turn on logging and
+    # -whatif switch to not actually make changes
+
+    # Path to the vendor configuration source file
+    [string]$PluginPath = "$env:CMDER_ROOT\vendor\sources-plugins.json",
+
+    # Replace the plugin
+    [switch]$force,
+
+    # Vendor folder location
+    [string]$saveTo = "$env:CMDER_ROOT\vendor\",
+
+    # Config folder location
+    [string]$config = "$env:CMDER_ROOT\config"
+)
+
+
+$ThisScript = Get-Item $MyInvocation.MyCommand.Definition
+
+# Dot source util functions into this scope
+. $($ThisScript.DirectoryName + "\utils.ps1")
+
+$bash_user_profile = $($env:cmder_root + "\config\user-profile.sh")
+$bash_user_aliases = $($env:cmder_root + "\config\user-aliases.sh")
+
+if (test-path $bash_user_aliases) {
+  $bash_aliases = $bash_user_aliases
+} else {
+  $bash_aliases = $bash_user_profile
+}
+
+$posh_user_profile = $($env:cmder_root + "\config\user-profile.ps1")
+$posh_user_aliases = $($env:cmder_root + "\config\user-aliases.ps1")
+
+if (test-path $posh_user_aliases) {
+  $posh_aliases = $posh_user_aliases
+} else {
+  $posh_aliases = $posh_user_profile
+}
+
+
+$cmder_user_profile = $($env:cmder_root + "\config\user-profile.cmd")
+$cmder_aliases = $($env:cmder_root + "\config\aliases")
```

Comment:
Rename what 'add-cmderplugin.ps1'? To what?  I think it is already pretty self explanatory what it is.